File: core/src/main/java/io/github/frikit/krandom/generator/user/Gender.java
/*
 * Copyright (c) 2026 krandom contributors
 *
 * Licensed under the MIT License. See LICENSE in the project root for license information.
 */
package io.github.frikit.krandom.generator.user;

/**
 * Biological sex / gender used to select locale-appropriate first names.
 *
 * <p>Pass to {@link FirstNameGenerator#generate(Gender)} to restrict output to names
 * traditionally associated with that gender in the configured locale.
 */
public enum Gender {
    MALE,
    FEMALE
}

File: core/src/main/kotlin/org/github/krandom/user/Gender.kt
/*
 * Copyright (c) 2026 krandom contributors
 *
 * Licensed under the MIT License. See LICENSE in the project root for license information.
 */
package org.github.krandom.user

import org.github.krandom.user.common.GenericUserGenerator

class Gender(extraGenders: List<String> = emptyList()) : KRandomUser<String>, GenericUserGenerator<String>() {

    companion object {
        val DEFAULT = listOf("Male", "Female")
    }

    //cache parsed list
    private val list = DEFAULT + extraGenders

    override fun randomData(): String {
        return randomData(list)
    }

    override fun randomDatas(size: Int): List<String> {
        return randomDatas(list, size)
    }
}
